maas/resource_maas_user.go

```diff
@@ -12,7 +12,7 @@ import (
 
 func resourceMAASUser() *schema.Resource {
 	return &schema.Resource{
-		Description:   "Provides a resource to manage MAAS users.",
+		Description:   "Provides a resource to manage MAAS users. *Note* You cannot use this to modify the logged in terraform user.",
```

Comment:
Maybe it makes sense to refer to non-local users that also cannot be managed by Terraform